[DOC] Document setting environment variable

pavithraes opened the pull request:
Fixes #759

<!-- Reference the issue corresponding to this PR. If an issues does not exist, consider opening one or writing a detailed description for what this PR changes and it's value in the following sections. -->
<!-- GitHub Docs on Keywords: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests -->

## Description

<!-- What is the purpose of this pull request? -->

This pull request:

- Adds a note about setting environment variable
- Minor misc clean-up

## Pull request checklist

<!-- Quick checklist to ensure high-quality Pull Request. -->

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

## Additional information

<!-- Do you have any other information about this pull request? This may include screenshots, references, and/or implementation notes. -->

Merge **after** the conda-store-ui feature is merged.

nkaretnikov commented on docusaurus-docs/conda-store-ui/tutorials/create-envs.md:
```diff
@@ -83,6 +83,21 @@ To install packages published only on [PyPI][pypi] using [`pip`][pip], include a
+### Set environment variables
+
+:::note
+This features is in the release *after* conda-store version 2024.1.1.
```
```suggestion
This feature is available *after* conda-store-ui version 2024.1.1.
```

nkaretnikov commented on docusaurus-docs/conda-store-ui/tutorials/setup.md:
```diff
@@ -47,11 +47,9 @@ conda-store-server --standalone
 
```
Please move changes in this file into a separate PR. To me, these look unrelated to the environment variables change, which means they can be merged separately (and likely much faster as well).

pavithraes replied:
Thanks for your review :)

I did mention in the PR description that it includes some minor, miscellaneous clean-up, so I thought this would be alright. 

However, I do agree we should aim for cleaner+specific PRs. I'll revert these updates.

nkaretnikov commented on docusaurus-docs/conda-store-ui/tutorials/create-envs.md:
````diff
@@ -83,6 +83,21 @@ To install packages published only on [PyPI][pypi] using [`pip`][pip], include a
+
+```yaml
+variables:
+   CONDA_OVERRIDE_CUDA: 12
````
```suggestion
    CONDA_OVERRIDE_CUDA: '12.0'
```

I suggest making this change because it makes it clear that the value is a string.

nkaretnikov commented on docusaurus-docs/conda-store-ui/tutorials/create-envs.md:
```diff
@@ -83,6 +83,21 @@ To install packages published only on [PyPI][pypi] using [`pip`][pip], include a
+:::note
+This features is in the release *after* conda-store version 2024.1.1.
+
+Currently, only the `CONDA_OVERRIDE_CUDA` environment variable is supported to set the CUDA version for GPU-powered workflows.
```
I'd link to the official conda documentation here as well, see:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html#overriding-detected-packages

nkaretnikov replied:
```suggestion
Currently, only the `CONDA_OVERRIDE_CUDA` environment variable can be specified, which allows setting the CUDA version for building packages with GPU support.
```
Proposed alternative wording, which is a bit more specific.

nkaretnikov reviewed (changes requested):
@pavithraes Thanks for documenting this! I've left some comments for your consideration.

Note: this feature is already in conda-store-server, but it's still in review in conda-store-ui. So I'm going to mark this PR as blocked until that one lands, see: https://github.com/conda-incubator/conda-store-ui/pull/354.

pavithraes commented:
@nkaretnikov Thanks for your review, I have addressed all your comments. :)

Please take another look and approve it when you get a chance. Of course, even after approval, we'll only merge after `https://github.com/conda-incubator/conda-store-ui/pull/354` is merged.

nkaretnikov commented on docusaurus-docs/conda-store-ui/tutorials/create-envs.md:
```diff
@@ -83,6 +83,22 @@ To install packages published only on [PyPI][pypi] using [`pip`][pip], include a
+Learn more in the [conda documentation][conda-docs-override-packages]
+:::
+
+You can set environment variables in the YAML-editor with the following syntax:
```
```suggestion
You can set environment variables in the YAML editor with the following syntax:
```

nkaretnikov replied:
This is just to have consistent spelling, it's "YAML editor" in other parts of the doc.

nkaretnikov commented on docusaurus-docs/conda-store-ui/tutorials/create-envs.md:
````diff
@@ -83,6 +83,22 @@ To install packages published only on [PyPI][pypi] using [`pip`][pip], include a
+
+```yaml
+variables:
+    CONDA_OVERRIDE_CUDA: '12.0'
````
```suggestion
  CONDA_OVERRIDE_CUDA: '12.0'
```

nkaretnikov replied:
This is just because our YAML pretty-printer uses 2 spaces by default.

nkaretnikov reviewed (approved):
@pavithraes Thanks, LGTM! I've left two comments, which would be nice to fix, but those are not blockers. OK to merge this after the companion UI PR lands: https://github.com/conda-incubator/conda-store-ui/pull/354
